### ISSUE TITLE
track_process netlink error

### ISSUE THREAD

**Issue by abelbeck** [acassen/keepalived#1099]
On Linux 3.16.61, Keepalived v2.0.11 generates this error, but seems to work:
```
Jan  7 09:29:04 pbx daemon.info Keepalived[4477]: Failed to open process monitoring socket - errno 93 - Protocol not supported
```

Relating to this code in `keepalived/core/track_process.c`:
```
static int
nl_connect(void)
{
    int rc;
    int nl_sock;
    struct sockaddr_nl sa_nl;

    nl_sock = socket(PF_NETLINK, SOCK_DGRAM | SOCK_CLOEXEC | SOCK_NONBLOCK, NETLINK_CONNECTOR);
    if (nl_sock == -1) {
        log_message(LOG_INFO, "Failed to open process monitoring socket - errno %d - %m", errno);
        return -1;
    }
```

More detail:
```
Jan  7 09:29:04 pbx daemon.info Keepalived[4477]: Starting Keepalived v2.0.11 (01/06,2019), git commit v2.0.10-75-g40afb8e+
Jan  7 09:29:04 pbx daemon.info Keepalived[4477]: Running on Linux 3.16.61-astlinux #1 SMP PREEMPT Sun Jan 6 18:09:03 CST 2019 (built for Linux 3.16.43)
Jan  7 09:29:04 pbx daemon.info Keepalived[4477]: Command line: 'keepalived' '-f' '/etc/keepalived/keepalived.conf'
Jan  7 09:29:04 pbx daemon.info Keepalived[4477]: Opening file '/etc/keepalived/keepalived.conf'.
Jan  7 09:29:04 pbx daemon.info Keepalived[4477]: Failed to open process monitoring socket - errno 93 - Protocol not supported
Jan  7 09:29:04 pbx daemon.info Keepalived[4480]: Starting VRRP child process, pid=4481
```

**Comment by pqarmitage** [comment]
Can you check the setting of CONFIG_PROC_EVENTS in the kernel config file. If it is not 'y', then your kernel doesn't support it.

**Comment by abelbeck** [comment]
Indeed, our kernel does not enable CONFIG_CONNECTOR -> CONFIG_PROC_EVENTS
```
# zcat /proc/config.gz | grep CONFIG_CONNECTOR
# CONFIG_CONNECTOR is not set
```
Is there a build-time way to disable track_process, or is it worth enabling this in the kernel ?

**Comment by pqarmitage** [comment]
I'll add a way of disabling it in the build, and I'll also update the error message to be more helpful.

If you want to be able to track processes without having to periodically run a script to do so, then it is worth enabling it in your kernel. It is also more responsive than running a script since keepalived is notified immediately a process is created or terminates, rather than waiting for the script to next run.

**Comment by pqarmitage** [comment]
@abelbeck Many thanks for pointing out the copy/paste error. All now merged.

**Comment by abelbeck** [comment]
@pqarmitage FYI, I tested your merged commit patch with Keepalived v2.0.11 and `--disable-track-process` works as expected.

Many thanks.
